src/core/reference/include/openvino/reference/gather.hpp

```diff
@@ -18,8 +18,13 @@ void gather(const T* const data,
             const Shape& data_shape,
             const Shape& indices_shape,
             const Shape& out_shape,
-            size_t axis,
+            int64_t generic_axis,
             size_t batch_dims = 0) {
+    // calculate real axis value
+    while (generic_axis < 0) {
+        generic_axis += data_shape.size();
+    }
+    auto axis = static_cast<size_t>(generic_axis);
```

Comment:
Apply it before reference function call (will produce smaller code as this normalization will applied for each instance.

There are util functions  to do it (with or without exception), also the gather attribute may have special value axis not set which probably should trigger some kind of assert.

Keep reference simple which assume it gets correct axis value.

Reply:
@mitruska Thanks for the explanation. 
Changed that back.